docs/src/references/mappers/c8y-mapper.md

```diff
@@ -568,6 +568,247 @@ c8y/alarm/alarms/create
 
 </div>
 
+### Twin
+
+The `twin` metadata is mapped to [inventory data](https://cumulocity.com/guides/concepts/domain-model/#inventory) in Cumulocity.
+
+#### Twin - Main device
+
+A device's digital twin model can be updated by publishing to a specific topic.
+
+The type part of the topic is used to group the data so it is easier for components to subscribe to relevant parts.
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/main///twin/device_OS
+```
+
+```json5 title="Payload"
+{
+  "family": "Debian",
+  "version": "11"
+}
+```
+
+</div>
+
+<div class="code-indent-right">
+
+**Cumulocity IoT (output)**
+
+```text title="Topic"
+c8y/inventory/managedObjects/update/<main-device-id>
+```
+
+```json5 title="Payload"
+{
+  "device_OS": {
+    "family": "Debian",
+    "version": "11"
+  }
+}
+```
+
+</div>
+
+
+#### Twin data - root
+
+Data can be added on the root level of the twin by publishing the values directly to the topic with the key used as type.
+The payload can be any valid JSON value other than a JSON object.
+JSON objects must be published to their typed topics directly.
+
+<div class="code-indent-left">
+
+**Thin Edge (input)**
+
+```text title="Topic (retain=true)"
+te/device/main///twin/subtype
+```
+
+```json5 title="Payload"
+"my-custom-type"
+```
+
+</div>
+
+<div class="code-indent-right">
+
+**Cumulocity IoT (output)**
+
+```text title="Topic"
+c8y/inventory/managedObjects/update/<main-device-id>
+```
+
+```json5 title="Payload"
+{
+  "subtype": "my-custom-type"
+}
+```
+
+</div>
+
+:::warning
+Updating the following properties via the `twin` channel is not supported
+
+* `name`
```

Comment:
Why is the `name` property forbidden? Looking at the code, `c8y_Agent` is the fragment that must be protected from overriding.

Reply:
The `name` and `type` fields are forbidden here because they're already part of the entity registration message. If someone wants to update these, they must send another registration message with the updated values. Allowing them to update the same data via both these endpoints will might be confusing especially in the broker as we may have different `name` and `type` in the device registration topic as well as in the `twin/...` topics.

Reply:
> The `name` and `type` fields are forbidden here because they're already part of the entity registration message. If someone wants to update these, they must send another registration message with the updated values. Allowing them to update the same data via both these endpoints will might be confusing especially in the broker as we may have different `name` and `type` in the device registration topic as well as in the `twin/...` topics.

Okay

> I didn't really understand why you think `c8y_Agent` must not be overridden?

What is a user publish its own fragment on `te/device/main///twin/c8y_Agent` ?

Reply:
> What if a user publish its own fragment on te/device/main///twin/c8y_Agent

I don't really have a use-case for someone wanting to update that fragment for the thin-edge device, as we set that. But I don't have any strong arguments to close that option either. So, will need @reubenmiller 's advise on this. For child devices, we probably wouldn't want to prevent them from setting it. For services, this fragment probably doesn't even make any sense.

Reply:
> > What if a user publish its own fragment on te/device/main///twin/c8y_Agent

> I don't really have a use-case for someone wanting to update that fragment for the thin-edge device, as we set that. But I don't have any strong arguments to close that option either. So, will need @reubenmiller 's advise on this. For child devices, we probably wouldn't want to prevent them from setting it. For services, this fragment probably doesn't even make any sense.

Indeed, this would make no sense, but could be done by a user who fails to notice that this is *already* done by thin-edge as for the `type`. I that case there will be inconsistencies.